main.go

```diff
@@ -507,6 +510,16 @@ func deployGoogleEndpointsServiceIfRequired(ctx context.Context, params Params)
 	}
 }
 
+func failIfCreatingNewPublicService(ctx context.Context, params Params, templateData TemplateData, name, namespace string) {
+	if params.Kind == "deployment" && params.Visibility == "public" {
+		serviceType, err := foundation.GetCommandWithArgsOutput(ctx, "kubectl", []string{"get", "service", name, "-n", namespace, "-o=jsonpath={.spec.type}"})
+		// fail if creating new public service
+		if err != nil || serviceType != "LoadBalancer" {
```

Comment:
I think we should handle the error differently. I think it should still fail/abort, but with a different message showing the error.

Reply:
Ok, what do you think we should phrase error message as?

Reply:
Maybe something like `log.Fatal().Err(err).Msgf("Failed to check current visibility. Visibility public is deprecated, please use esp or apigee.")`

Reply:
Updated, please check. err may not be always not-nil, so I have not added it.

Reply:
Well, that was kind of the idea actually :sweat_smile: . To handle the `err` differently, like this:
```
if err != nil {
  log.Fatal().Err(err).Msgf("Failed to check current visibility. Visibility public is deprecated, please use esp or apigee.")
}
if serviceType != "LoadBalancer" {
  log.Fatal().Msgf("Creating new public service or changing to visibility public is no longer supported, please use visibility esp or apigee.")
}
```

Reply:
I wanted to avoid two checks and messages as there will be error mostly when you try to create new service. So I wanted to avoid that. But anyways updated it.

Reply:
I understand, but if there is another reason for error (e.g. connection issues), it would be really hard to find the problem.